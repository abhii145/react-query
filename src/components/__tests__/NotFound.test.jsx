import { render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import NotFound from "../NotFound"
import { it, expect } from "vitest"

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <NotFound />
    </MemoryRouter>
  )
  expect(asFragment()).toMatchSnapshot()
})

it("renders NotFound component", () => {
  render(
    <MemoryRouter>
      <NotFound />
    </MemoryRouter>
  )

  expect(screen.getByText("404")).toBeInTheDocument()
  expect(screen.getByText("Page not found")).toBeInTheDocument()
  expect(
    screen.getByText("Sorry, we couldn’t find the page you’re looking for.")
  ).toBeInTheDocument()
  expect(
    screen.getByRole("link", { name: /Go back home/i })
  ).toBeInTheDocument()
})
