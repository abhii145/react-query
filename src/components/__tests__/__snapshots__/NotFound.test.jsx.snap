// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-gray-50 min-h-screen flex flex-col items-center justify-center p-6"
  >
    <div
      class="max-w-2xl text-center p-8 rounded-lg shadow-md"
    >
      <div
        class="text-center"
      >
        <p
          class="text-base font-semibold text-indigo-600"
        >
          404
        </p>
        <h1
          class="mt-4 text-balance text-5xl font-semibold tracking-tight text-gray-900 sm:text-7xl"
        >
          Page not found
        </h1>
        <p
          class="mt-6 text-pretty text-lg font-medium text-gray-500 sm:text-xl/8"
        >
          Sorry, we couldn’t find the page you’re looking for.
        </p>
        <div
          class="mt-10 flex items-center justify-center gap-x-6"
        >
          <a
            class="rounded-md bg-indigo-600 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
            data-discover="true"
            href="/"
          >
            Go back home
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
